lab_3_generate_by_ngrams/main.py

```diff
@@ -331,6 +481,19 @@ def continue_sequence(
 
         In case of corrupt input arguments or unexpected behaviour of methods used return None.
         """
+        if not (isinstance(sequence, tuple) and sequence
+                and isinstance(next_tokens, list) and next_tokens
+                and isinstance(sequence_candidates, dict)
+                and sequence_candidates
+                and len(next_tokens) <= self._beam_width
+                and sequence in sequence_candidates):
+            return None
+        for token_tuple in next_tokens:
+            new_sequence = sequence + (token_tuple[0],)
+            new_freq = sequence_candidates[sequence] - math.log(token_tuple[1])
+            sequence_candidates[new_sequence] = new_freq
+        sequence_candidates.pop(sequence)
+        return sequence_candidates
```

Comment:
Beware of mutation!!!
Dictionaries are mutable, thus you are changing this dictionary outside of this function as well, which is undesired.

Reply:
I tried to use copy() to avoid mutation but it didn't work. Could you give me a hint how to solve this problem?

Reply:
`copy()` is the way to go, what do you mean it didn't work?

Reply:
It started not to pass one of the tests